File: 2904-shortest-and-lexicographically-smallest-beautiful-string/2904-shortest-and-lexicographically-smallest-beautiful-string.py
class Solution:
    def shortestBeautifulSubstring(self, s: str, k: int) -> str:
        n = len(s)
        prefix = [0]*(n+1)
        for i in range(1,n+1):
            if s[i-1] == '1':
                prefix[i] = prefix[i-1] + 1
            else:
                prefix[i] = prefix[i-1]
        
        res = ""
        for j in range(1, n+1):
            for i in range(j):
                if prefix[j]-prefix[i] == k:
                    if (not res) or j-i < len(res):
                        res = s[i:j]
                    elif j-i == len(res) and s[i:j] < res:
                        res = s[i:j]
        return res

File: 2904-shortest-and-lexicographically-smallest-beautiful-string/2904-shortest-and-lexicographically-smallest-beautiful-string.cpp
class Solution {
public:
    string shortestBeautifulSubstring(string s, int k) {
        int n = s.size();
        vector<string>ans;
        int  mini = 1e9;
        for(int i=0; i<n; i++) {
            int count = 0;
            string temp;
            for(int j = i; j<n; j++) {
                if(s[j]=='1') count++;
                if(count<=k) temp+=s[j];
                if(count==k){
                    ans.push_back(temp);
                    mini = min(mini, j-i+1);
                }
            }
        }
        sort(ans.begin() , ans.end());
        for(auto x: ans) {
            if(x.size()==mini) return x;
        }
        return "";
    }
};